scipy/stats/_distn_infrastructure.py

```diff
@@ -2652,16 +2654,16 @@ def entropy(pk, qk=None, base=None):
 
     """
     pk = asarray(pk)
-    pk = 1.0*pk / np.sum(pk, axis=0)
+    pk = 1.0*pk / np.sum(pk, axis=axis, keepdims=True)
     if qk is None:
         vec = entr(pk)
     else:
         qk = asarray(qk)
         if len(qk) != len(pk):
```

Comment:
This check will need to be updated. I assume this should now be `qk.shape != pk.shape`?

Reply:
Agreed, I also added a test for this